src/Protego.t.sol

```diff
@@ -65,337 +47,157 @@ interface DSTokenLike {
+        assertFalse(protego.planned(usr, tag, sig, eta), "After drop spell: spell still planned");
     }
 
     // Test drop anything by lifting Protego to hat
-    function testDropAllSpells() external {
-
-        uint256 iter = 100;
+    function testDropAllSpellsParams() public {
+        uint256 iter = 10;
         BadSpells[] memory badSpells = new BadSpells[](iter);
 
-        for (uint i = 0; i < iter; i++) {
-            DssEndTestSpell badSpell = new DssEndTestSpell(address(pause), address(end));
-
-            vote(address(badSpell));
-
+        for (uint256 i = 0; i < iter; i++) {
+            ConformingSpell badSpell = new ConformingSpell(pause, end);
+            _vote(address(badSpell));
             badSpell.schedule();
 
-            badSpells[i].badSpell = address(badSpell);
-
-            assertTrue(protego.planned(DSSpellLike(address(badSpell))));
-        }
-
-        vote(address(protego));
-
-        for (uint i = 0; i < iter; i++) {
-
-            protego.drop(DSSpellLike(badSpells[i].badSpell));
+            badSpells[i].action = badSpell.action();
+            badSpells[i].tag = badSpell.tag();
+            badSpells[i].sig = badSpell.sig();
+            badSpells[i].eta = badSpell.eta();
 
-            assertTrue(!protego.planned(DSSpellLike(badSpells[i].badSpell)));
+            assertTrue(protego.planned(badSpells[i].action, badSpells[i].tag, badSpells[i].sig, badSpells[i].eta));
         }
-    }
 
-    // Test drop anything by lifting Protego to hat
-    function testDropAllSpellsParams() external {
-
-        uint256 iter = 100;
-        BadSpells[] memory badSpells = new BadSpells[](iter);
-
-        for (uint i = 0; i < iter; i++) {
-            DssEndTestSpell badSpell = new DssEndTestSpell(address(pause), address(end));
-
-            vote(address(badSpell));
-
-            badSpell.schedule();
+        _vote(address(protego));
 
-            badSpells[i].action   = badSpell.action();
-            badSpells[i].tag      = badSpell.tag();
-            badSpells[i].sig      = badSpell.sig();
-            badSpells[i].eta      = badSpell.eta();
+        for (uint256 i = 0; i < iter; i++) {
+            vm.expectEmit(true, true, true, true);
+            emit Drop(protego.id(badSpells[i].action, badSpells[i].tag, badSpells[i].sig, badSpells[i].eta));
+            protego.drop(badSpells[i].action, badSpells[i].tag, badSpells[i].sig, badSpells[i].eta);
 
-            assertTrue(protego.planned(badSpells[i].action, badSpells[i].tag, badSpells[i].sig, badSpells[i].eta));
+            assertFalse(protego.planned(badSpells[i].action, badSpells[i].tag, badSpells[i].sig, badSpells[i].eta));
         }
 
-        vote(address(protego));
+        // After Protego loses the hat, it can no longer drop spells
+        _vote(address(0));
+        vm.expectRevert();
+        protego.drop(badSpells[0].action, badSpells[0].tag, badSpells[0].sig, badSpells[0].eta);
```

Comment:
Is it `ds-pause-proxy-unauthorized` with which it fails? Then, could you please specify it in `vm.expectRevert("ds-pause-proxy-unauthorized");`

Reply:
Do you mean `ds-auth-unauthorized`? I'll add that for explicitness.

Reply:
Done.